mycore-restapi/src/main/java/org/mycore/restapi/MCRNormalizeMCRObjectIDsFilter.java

```diff
@@ -0,0 +1,163 @@
+import org.mycore.solr.MCRSolrClientFactory;
+
+/**
+ * This prematching filter checks the given MCRObjectIDs in an REST API call beginning with /objects,
+ * normalizes them and sends a redirect if necessary.
+ * 
+ * @author Robert Stephan
+ *
+ */
+@Provider
+@PreMatching
+public class MCRNormalizeMCRObjectIDsFilter implements ContainerRequestFilter {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private static List<String> SEARCHKEYS_FOR_OBJECTS = Arrays.asList(
+            MCRConfiguration2.getString("MCR.RestAPI.V2.AlternativeIdentifier.Objects.Keys").orElse("").split(","));
+    private static List<String> SEARCHKEYS_FOR_DERIVATES = Arrays.asList(
+            MCRConfiguration2.getString("MCR.RestAPI.V2.AlternativeIdentifier.Derivates.Keys").orElse("").split(","));
+
+    @Context
+    ResourceInfo resourceInfo;
+
+    @Context
+    HttpServletResponse response;
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        UriInfo uriInfo = requestContext.getUriInfo();
+        String path = uriInfo.getPath().toString();
+        String[] pathParts = path.split("/");
+        if (pathParts.length >= 2 && "objects".equals(pathParts[0])) {
+            String mcrid = pathParts[1];
+            try {
+                if (mcrid.contains(":")) {
+                    String key = mcrid.substring(0, mcrid.indexOf(":"));
+                    String value = mcrid.substring(mcrid.indexOf(":") + 1);
+                    if (SEARCHKEYS_FOR_OBJECTS.contains(key)) {
+                        ModifiableSolrParams params = new ModifiableSolrParams();
+                        params.set("start", 0);
+                        params.set("rows", 1);
+                        params.set("fl", "id");
+                        params.set("fq", "objectKind:mycoreobject");
+                        params.set("q", key + ":" + ClientUtils.escapeQueryChars(value));
+                        QueryResponse solrResponse;
+                        try {
+                            solrResponse = MCRSolrClientFactory.getMainSolrClient().query(params);
+                            SolrDocumentList solrResults = solrResponse.getResults();
+                            if (solrResults.getNumFound() >= 1) {
+                                mcrid = String.valueOf(solrResults.get(0).getFieldValue("id"));
+                                pathParts[1] = mcrid;
+                            }
+                        } catch (Exception e) {
+                            LOGGER.error("Error retrieving object id from SOLR", e);
+                        }
+                    }
+                } else {
+                    MCRObjectID mcrObjID = MCRObjectID.getInstance(mcrid);
+                    // set the properly formated mcrObjID back to URL
+                    mcrid = mcrObjID.toString();
+                    if (!mcrid.equals(pathParts[1])) {
+                        pathParts[1] = mcrid;
```

Comment:
the check for equals is not needed here, as assigning an array element is far less complex than checking for equal

Reply:
O.K.